source/MaterialXCore/Interface.cpp

```diff
@@ -334,7 +334,7 @@ bool Input::validate(string* message) const
     {
         validateRequire(getDefaultGeomProp() != nullptr, res, message, "Invalid defaultgeomprop string");
     }
-    if (parent->isA<Node>())
+    else if (parent->isA<Node>())
```

Comment:
Wouldn't this change introduce a subtle bug, where a node input with a `defaultgeomprop` string but no binding would now be considered valid?

My thinking here is that only `nodedef` inputs support `defaultgeomprop` attributes in MaterialX, so the presence of this attribute on a _node_ input should have no effect,.

Here's the section of the specification that discusses `defaultgeomprop` attributes in the context of `nodedef` inputs, and let me know if I'm missing a section that might discuss additional use cases:

https://materialx.org/assets/MaterialX.v1.38.Spec.pdf#page=51

Reply:
Yes, that is the correct intended interpretation: defaults, including "defaultgeomprop"s, can only be defined on nodedefs, not in node instances.

Reply:
This should be fixed now. Please have a look. I added in a extra check that the parent must be a `<nodedef>` for an input to have a `defaultgeomprop`. Thanks.

Reply:
I've added in what should be the proper test now and fix a failing unit test.